sklearn/utils/estimator_checks.py

```diff
@@ -2416,6 +2417,69 @@ def check_classifiers_regression_target(name, estimator_orig):
         assert_raises_regex(ValueError, msg, e.fit, X, y)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_estimator_sparse_dense(name, estimator_orig):
+    rng = np.random.RandomState(52)
+    estimator = clone(estimator_orig)
+    estimator_sp = clone(estimator_orig)
+    tags = _safe_tags(estimator_orig)
+    centers = 2 if tags["binary_only"] else None
+    X, y = make_blobs(random_state=rng, cluster_std=0.5, centers=centers)
+    # for put some points to zero to have a little bit of sparsity
+    if tags["requires_positive_X"]:
+        X -= np.min(X) - 1.
+    X_csr = sparse.csr_matrix(X)
+    if tags["multioutput"] or tags["multioutput_only"]:
+        y = y[:, np.newaxis]
+
+    for sparse_format in ['csr', 'csc', 'dok', 'lil', 'coo', 'dia', 'bsr']:
+        X_sp = X_csr.asformat(sparse_format)
+        # catch deprecation warnings
+        with ignore_warnings(category=DeprecationWarning):
+            if name in ['Scaler', 'StandardScaler']:
+                estimator.set_params(with_mean=False)
+                estimator_sp.set_params(with_mean=False)
+            if name in ['RobustScaler']:
+                # XXX naming is not consistent (with_mean vs with_centering) :(
```

Comment:
Should I open an issue for that ?

Reply:
yes +1 so at we can discuss it. Once the issue is opened you can remove this comment.

Reply:
Done: #14810